builder/src/components/PackageManagement/Context.tsx

```diff
@@ -0,0 +1,57 @@
+import { PackageCategory } from "../../api";
+import React, { PropsWithChildren, useContext, useState } from "react";
+
+export type ContextProps = {
+    isDeprecated: boolean;
+    canDeprecate: boolean;
+    canUndeprecate: boolean;
+    canUnlist: boolean;
+    canUpdateCategories: boolean;
+    csrfToken: string;
+    currentCategories: PackageCategory[];
+    availableCategories: PackageCategory[];
+    packageListingId: string;
+};
+
+export interface IManagementContext {
+    props: ContextProps;
+    setCategories: (categories: PackageCategory[]) => void;
+    closeModal: () => void;
+}
+
+export interface ManagementContextProviderProps {
+    initial: ContextProps;
+    closeModal: () => void;
+}
+
+export const ManagementContextProvider: React.FC<
+    PropsWithChildren<ManagementContextProviderProps>
+> = ({ children, initial, closeModal }) => {
+    const [state, setState] = useState<ContextProps>(initial);
+
+    const setCategories = (categories: PackageCategory[]) => {
+        setState({
+            ...state,
+            currentCategories: categories,
+        });
+    };
+
+    return (
+        <ManagementContext.Provider
+            value={{
+                props: state,
+                setCategories,
+                closeModal,
+            }}
+        >
+            {children}
+        </ManagementContext.Provider>
+    );
+};
+export const ManagementContext = React.createContext<
+    IManagementContext | undefined
+>(undefined);
+
+export const useManagementContext = (): IManagementContext => {
```

Comment:
Non-null assertion is often avoided with something like this:

```typescript
    const context = useContext(ManagementContext);

    if (context === undefined) {
        throw new Error("useManagementContext must be used inside a ManagementContextProvider");
    }

    return context;
```

Reply:
Intentionally left out since I didn't see this ever actually solving anything, really it just converts one kind of an error to another kind of an error in a situation which should never occur in the first place.